src/classes/STG_PanelOppNaming_CTRL.cls

```diff
@@ -0,0 +1,222 @@
+                if (string.isBlank(ons.Opportunity_Record_Types__c))
+                    ons.Opportunity_Record_Types__c = Label.stgLabelAllRecordTypes;
+            }
+            return listOppNameSettings;
+        }
+        set;
+    }
+
+    /*********************************************************************************************************
+    * @description The panel's constructor 
+    */
+    public STG_PanelOppNaming_CTRL() {
+        isRunningBatch = false;
+    }
+        
+    /*********************************************************************************************************
+    * @description Returns the string Id of the Households panel. 
+    */
+    public override string idPanel() { return 'idPanelOppNaming'; }
+
+    /*********************************************************************************************************
+    * @description Whether we are running the batch opportunity naming process 
+    */
+    public boolean isRunningBatch { get; set; }
+
+    /*********************************************************************************************************
+    * @description Action Method to delete a specific settings instance.
+    * @return null 
+    */
+    public PageReference delONS() {
+        delete mapOppNameSettings.get(currentONSName);
+        mapOppNameSettings = Opportunity_Naming_Settings__c.getAll();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to enter Edit mode for a specific setting instance.
+    * @return null 
+    */
+    public PageReference editONS() {
+        currentONS = mapOppNameSettings.get(currentONSName);
+        super.editSettings();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to create a new settings object in memory, then enter Edit mode.
+    * @return null 
+    */
+    public PageReference newONS() {
+        currentONS = new Opportunity_Naming_Settings__c(
+            Name = System.now().format()
+        );
+        super.editSettings();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to save the current edited setting, refresh the list of settings, and return
+    * to the main page.
+    * @return null 
+    */
+    public PageReference saveONS() {
+        upsert currentONS;
+        mapOppNameSettings = Opportunity_Naming_Settings__c.getAll();
+        super.cancelEdit();
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Action Method to run batch opportunity renaming.
+    * @return null 
+    */
+    public PageReference runBatch() {
```

Comment:
does running the batch need to save any current ons edit?  I haven't seen the ui, so I don't know if this can happen or not.  I know in household naming, it can be an issue.